programs/ssl/ssl_client2.c

```diff
@@ -1971,7 +2008,16 @@ int main(int argc, char *argv[])
     }
 
 #if defined(MBEDTLS_SSL_EARLY_DATA)
-    mbedtls_ssl_conf_early_data(&conf, opt.early_data);
+    int early_data_enabled = 0;
+    size_t early_data_len;
+    if (strlen(opt.early_data_file) > 0 &&
+        ssl_early_data_read_file(opt.early_data_file,
+                                 &early_data, &early_data_len) == 0) {
+        early_data_enabled = MBEDTLS_SSL_EARLY_DATA_ENABLED;
+    } else {
+        early_data_enabled = MBEDTLS_SSL_EARLY_DATA_DISABLED;
+    }
+    mbedtls_ssl_conf_early_data(&conf, early_data_enabled);
```

Comment:
See also above suggestion.
```suggestion
    mbedtls_ssl_conf_early_data(&conf, early_data_fp == NULL ? MBETLS_SSL_EARLY_DATA_DISABLED : MBETLS_SSL_EARLY_DATA_ENABLED );
```

Reply:
Simplify the file read and change it, a little different from this.